edk2toollib/uefi/edk2/path_utilities.py

```diff
@@ -116,34 +131,56 @@ def GetAbsolutePathOnThisSytemFromEdk2RelativePath(self, relpath):
     def GetContainingPackage(self, InputPath):
         self.logger.debug("GetContainingPackage: %s" % InputPath)
 
+        # Make a list that has the path case normalized for comparison.
+        # This only does anything on Windows
+        NormCasePackagesPathList = [os.path.normcase(x) for x in self.PackagePathList]
+
+        # check InputPath to make sure it is at least in the folder structure of the code tree
+        if os.path.normcase(self.WorkspacePath) not in os.path.normcase(InputPath):
+            # not in workspace - check all the packages paths
+            found_in_pp = False
+            for p in NormCasePackagesPathList:
+                if p in os.path.normcase(InputPath):
+                    found_in_pp = True
+                    break
+            if(not found_in_pp):
+                self.logger.error(f"{InputPath} not in code tree")
+                self.logger.info("PackagePath is: %s" % os.pathsep.join(self.PackagePathList))
+                self.logger.info("Workspace path is : %s" % self.WorkspacePath)
+                return None
+
+        # InputPath is in workspace or PackagesPath for worst case scenario.
+
         dirpathprevious = os.path.dirname(InputPath)
         dirpath = os.path.dirname(InputPath)
-        while(dirpath is not None):
+        while(True):
```

Comment:
I would like some other termination condition

Reply:
I agree but i couldn't find a condition and during unit test I could not execute failed path.  If you look at code coverage you will that is the only code not executed.  

I guess i could have a counter to avoid an infinite loop.  Something like 200.   Do you like that better?  I still won't be able to execute it as i don't think it is possible with path length limits.

Reply:
@corthon and @matthewfcarlson  waiting on ideas if this is worth it?

Reply:
I'm fine with just a counter to avoid infinite loops

Reply:
please confirm you are ok with solution.